lib/puppet_x/puppetlabs/strings/yard/handlers/puppet_4x_function_handler.rb

```diff
@@ -33,22 +52,73 @@ class Puppet4xFunctionHandler < YARD::Handlers::Ruby::Base
     # Then, traverse the children of the parameters, storing each identifier
     # in the list of method arguments.
     obj = MethodObject.new(function_namespace, name) do |o|
+    end
+
+    # overload_signatures is a list of lists of tuples or empty:
+    # overload_signatures = [
+    #   [ # First function dispatch arguments
+    #     # argument name, argument type
+    #     ['arg0', 'Variant[String,Array[String]]'
+    #     ['arg1', 'Optional[Type]']
+    #   ],
+    #   [ # Second function dispatch arguments
+    #     ['arg0', 'Variant[String,Array[String]]'
+    #     ['arg1', 'Optional[Type]']
+    #     ['arg2', 'Any']
+    #   ]
+    # ]
+    overload_signatures = []
+    statement.traverse do |node|
+      # Find all of the dispatch methods
+      if node.type == :ident and node.source == 'dispatch'
+        command = node.parent
+        do_block = command.jump :do_block
+        # If the command doesn't have a do_block we can't extract type info
+        if do_block == command
+          next
+        end
+        # Iterate through each of the children of the do block and build
+        # tuples of parameter names and parameter type signatures
+        do_block.children.first.children.each do |child|
+          overload_signatures <<= [extract_type_from_command(child)]
+        end
+      end
+    end
+
+    # If the overload_signatures list is empty because we couldn't find any
+    # dispatch blocks, then there must be one function named the same as the
```

Comment:
"must be one method" instead of "must be one function" (would make the comment more clear).